tools/install_base.sh

```diff
@@ -61,18 +61,42 @@ escape_rhs_sed ()
 
 }
 
-escape_lhs_sed ()
+escape_lhs_sed()
 {
     echo $(printf '%s\n' "$1" | sed 's:[][\/.^$*]:\\&:g')
 
 }
 
-compute-version ()
+compute-version()
 {
     echo $(ls $1/dist/lib/scheduler-server-*.jar | sed "s/^$(escape_lhs_sed $1/dist/lib/scheduler-server-)\(.*\)\.jar$/\1/g")
 
 }
 
+compute-major-minor()
+{
+    NUMERIC_VERSION=$(echo ${1/-SNAPSHOT})
+    echo ${NUMERIC_VERSION%.*}
+}
+
+# an upgrade of the password stored format occurs when the new version is 14.1 or greater and the old version is 14.0 or lower
+is_legacy_hash_upgrade()
+{
+  NEW_VERSION_LEGACY_SETTING=$( grep 'pa.rm.legacy.encryption=' $PA_ROOT/default/config/rm/settings.ini | head -n 1 | cut -d '=' -f2 )
+  OLD_VERSION_LEGACY_SETTING=$( grep 'pa.rm.legacy.encryption=' $PA_ROOT/previous/config/rm/settings.ini | head -n 1 | cut -d '=' -f2 )
+  if [ "$NEW_VERSION_LEGACY_SETTING" == "" ]; then
+    NEW_VERSION_LEGACY_SETTING="true"
+  fi
+  if [ "$OLD_VERSION_LEGACY_SETTING" == "" ]; then
+    OLD_VERSION_LEGACY_SETTING="true"
+  fi
+  if [ "$NEW_VERSION_LEGACY_SETTING" == "false" ] & [ "$OLD_VERSION_LEGACY_SETTING" == "true" ]; then
+    return 0
+  else
```

Comment:
the else involves:
false && false
true && true
true && false
false && false
I don't think it is good ...

Reply:
we need to upgrade when the new version has pa.rm.legacy.encryption set to false and the old version has it set to true or absent

Reply:
legacy.encryption is set to true when we enable backward compatibility mode with the old encryption mechanism